### PR TITLE
CTCP-3885: add Navigation for place of loading

### PR DESCRIPTION
**Pull request by kenneth-rayner**

### CODE REVIEW

File: app/navigation/LoadingNavigator.scala [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import pages.Page
+import pages.loading._
+import play.api.mvc.Call
+
+@Singleton
+class LoadingNavigator {
+
+  protected def normalRoutes(departureId: String, mode: Mode): PartialFunction[Page, UserAnswers => Option[Call]] = {
+    case AddUnLocodeYesNoPage         => ua => addUnlocodeNormalRoute(ua, departureId)
+    case UnLocodePage                 => ua => CountryPage.route(ua, departureId, NormalMode)
```
Comment (kevincoatsworth) [inline app/navigation/LoadingNavigator.scala:30]: I think this should be go to the AddExtraInformationYesNo page - it's optional whether they add country/location

File: test/navigator/LoadingNavigatorSpec.scala [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+          val userAnswers = emptyUserAnswers.setValue(AddUnLocodeYesNoPage, false)
+          navigator
+            .nextPage(AddUnLocodeYesNoPage, userAnswers, departureId, mode)
+            .mustBe(CountryPage.route(userAnswers, departureId, mode).value)
+        }
+
+      }
+
+      "must go from UnLocodePage to CountryPage" - {
```
Comment (kevincoatsworth) [inline test/navigator/LoadingNavigatorSpec.scala:55]: These would need to tweaking too

**Review by kevincoatsworth — DISMISSED** [review]

File: app/controllers/loading/AddExtraInformationYesNoController.scala [hunk 3]
```diff
@@ -34,7 +34,7 @@ import scala.concurrent.{ExecutionContext, Future}
 class AddExtraInformationYesNoController @Inject() (
   override val messagesApi: MessagesApi,
   implicit val sessionRepository: SessionRepository,
-  navigator: Navigator,
+  navigator: LoadingNavigator,
```
Comment (justega247) [inline app/controllers/loading/AddExtraInformationYesNoController.scala:37]: What is the difference between this approach and using the annotation approach with a bound LoadingNavigator?

Comment (JakeGoodwin) [reply app/controllers/loading/AddExtraInformationYesNoController.scala:37]: Also interested in this but think we need to be consistent across both tickets. Ive noticed the other PR uses the annotation approach. @kenneth-rayner might be worth speaking with @BledarFeratiEY to keep both navigation tickets in line

**Review by JakeGoodwin — APPROVED** [review]

**Review by JakeGoodwin — APPROVED** [review]

**Review by jackheslop96 — APPROVED** [review]

**Pull request merged**